pkg/promql/engine.go

```diff
@@ -651,34 +692,72 @@ func (f inspector) Visit(node parser.Node, path []parser.Node) (parser.Visitor,
 	return f, nil
 }
 
-// This is modified from upstream to avoid descending into a MatrixSelector's VectorSelector
-func (ng *Engine) findMinTime(s *parser.EvalStmt) (time.Time, error) {
-	var maxOffset time.Duration
-
-	err := parser.Walk(inspector(func(node parser.Node, path []parser.Node) (bool, error) {
-		subqOffset, subqRange := ng.subqueryOffsetRange(path)
+func (ng *Engine) findMinMaxTime(s *parser.EvalStmt) (int64, int64) {
```

Comment:
with this change, I believe we can delete the inspector type above (line 685-693)